numpy/lib/_arraypad_impl.py

```diff
@@ -745,8 +749,39 @@ def pad(array, pad_width, mode='constant', **kwargs):
            [100, 100,   3,   4,   5, 100, 100],
            [100, 100, 100, 100, 100, 100, 100],
            [100, 100, 100, 100, 100, 100, 100]])
+
+    >>> a = np.arange(2 * 3).reshape(2, 3)
```

Comment:
It is confusing that `0` is both the pad value and one of the values in the original array. What about this? Or add a pad value?
```suggestion
    >>> a = np.arange(1, 7).reshape(2, 3)
```